hazelcast/src/main/java/com/hazelcast/logging/ILogger.java

```diff
@@ -205,17 +35,68 @@ public interface ILogger extends TpcLogger {
     void log(LogEvent logEvent);
 
     /**
-     * Gets the logging level.
-     *
-     * @return the logging level
-     */
-    Level getLevel();
-
-    /**
-     * Checks if a message at the given level is going to be logged by this logger.
-     *
-     * @param level the log level
-     * @return true if this logger will log messages for the given level, false otherwise
-     */
-    boolean isLoggable(Level level);
+     * Logs to {@link #fine(String)} using a lazily evaluated {@code template} {@link String} with arguments, formatted using
+     * {@link String#format(String, Object...)}
+     *
+     * @since 5.4
+     */
+    default void logFine(String template, Object arg1) {
+        if (isFineEnabled()) {
+            fine(String.format(template, arg1));
+        }
+    }
+
+    /** @see #logFine(String, Object) */
+    default void logFine(String template, Object arg1, Object arg2) {
+        if (isFineEnabled()) {
+            fine(String.format(template, arg1, arg2));
+        }
+    }
+
+    /** @see #logFine(String, Object) */
+    default void logFine(String template, Object arg1, Object arg2, Object arg3) {
+        if (isFineEnabled()) {
+            fine(String.format(template, arg1, arg2, arg3));
+        }
+    }
+
+    /** @see #logFine(String, Object) */
+    default void logFine(String template, Object arg1, Object arg2, Object arg3, Object arg4) {
+        if (isFineEnabled()) {
+            fine(String.format(template, arg1, arg2, arg3, arg4));
+        }
+    }
+
+    /**
+     * Logs to {@link #finest(String)} using a lazily evaluated {@code template} {@link String} with arguments, formatted using
+     * {@link String#format(String, Object...)}
+     *
+     * @since 5.4
+     */
+    default void logFinest(String template, Object arg1) {
```

Comment:
I 'd rather skip "log" in the method name to match existing methods' naming: `finest`, `fine` etc

Reply:
a547aba8cadcc6ed16fc3f41e1d1f2415cce159f